webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -12454,50 +12461,345 @@ const firstVisibleMsg = {
                 $mck_loc_lon.trigger('change');
                 if (CURR_LOC_ADDRESS) {
                     $mck_loc_address.val(CURR_LOC_ADDRESS);
-                } else if (GEOCODER) {
-                    var latlng = {
-                        lat: MCK_CURR_LATITIUDE,
-                        lng: MCK_CURR_LONGITUDE,
-                    };
-                    GEOCODER.geocode(
+                }
+                if (mapMarker) {
+                    syncMarkerPosition(
                         {
-                            location: latlng,
+                            lat: MCK_CURR_LATITIUDE,
+                            lng: MCK_CURR_LONGITUDE,
                         },
-                        function (results, status) {
-                            if (status === 'OK') {
-                                if (results[1]) {
-                                    CURR_LOC_ADDRESS = results[1].formatted_address;
-                                }
-                            }
+                        {
+                            pan: true,
                         }
                     );
                 }
+                reverseGeocodeLatLng({
+                    lat: MCK_CURR_LATITIUDE,
+                    lng: MCK_CURR_LONGITUDE,
+                });
             };
             _this.openMapBox = function () {
-                $mckMapContent.locationpicker({
-                    location: {
-                        latitude: MCK_CURR_LATITIUDE,
-                        longitude: MCK_CURR_LONGITUDE,
-                    },
-                    radius: 0,
-                    scrollwheel: true,
-                    inputBinding: {
-                        latitudeInput: $mck_loc_lat,
-                        longitudeInput: $mck_loc_lon,
-                        locationNameInput: $mck_loc_address,
-                    },
-                    enableAutocomplete: true,
-                    enableReverseGeocode: true,
-                    onchanged: function (currentLocation) {
-                        MCK_CURR_LATITIUDE = currentLocation.latitude;
-                        MCK_CURR_LONGITUDE = currentLocation.longitude;
-                    },
-                });
-                $mck_loc_box.on('shown.bs.mck-box', function () {
-                    $mckMapContent.locationpicker('autosize');
+                if (!w.google || !w.google.maps) {
```

Comment:
can you move all maps related code to a separate file
this sidebox file is becoming huge to manage.